prov/sockets/include/sock.h

```diff
@@ -905,15 +924,29 @@ enum {
 	SOCK_CONN_SHUTDOWN,
 };
 
+struct sock_conn_req_handle;
+typedef void (*sock_ep_cm_handle_event_fn) (struct sock_ep_cm_head *cm_head,
+                                            struct sock_conn_req_handle *handle);
+
 struct sock_conn_req_handle {
 	struct fid handle;
 	struct sock_conn_req *req;
 	int sock_fd;
-	int is_accepted;
+	union {
+		struct {
+			uint8_t monitored: 1;
+			uint8_t accepted:  1;
+			uint8_t finalized: 1;
+			uint8_t rejected:  1;
+			uint8_t closing:   1;
+		};
+	} flags;
```

Comment:
Looking at how these are used, we can make the following changes:
- monitored: This looks okay as a flag.
- accepted: This can be removed.  It's only used as a minor validation check against the user accepting then rejecting the same request.

The last 3 are independent and are better as states than flags.  A connection is either rejected or closing, but not both.  And finalized is reached by going through closing.  Accepted/active could be kept as another state used prior to moving to closing or rejected.

Reply:
Thanks for the review!
I removed the list of flags. Instead I used an enum to list all the possible states a connection handler can have.